docs/sources/flow/reference/components/prometheus.exporter.elasticsearch.md

```diff
@@ -0,0 +1,107 @@
+`node` | `string` | Node's name of which metrics should be exposed |  | no
+`indices` | bool | Export stats for indices in the cluster. |  | no
+`indices_settings` | bool | Export stats for settings of all indices of the cluster. |  | no
+`cluster_settings` | bool | Export stats for cluster settings. |  | no
+`shards` | bool | Export stats for shards in the cluster (implies indices). |  | no
+`snapshots` | bool | Export stats for the cluster snapshots. |  | no
+`clusterinfo_interval` | `duration` | Cluster info update interval for the cluster label. | `"5m"` | no
+`ca` | `string` | Path to PEM file that contains trusted Certificate Authorities for the Elasticsearch connection. |  | no
+`client_private_key` | `string` | Path to PEM file that contains the private key for client auth when connecting to Elasticsearch. |  | no
+`client_cert` | `string` | Path to PEM file that contains the corresponding cert for the private key to connect to Elasticsearch. |  | no
+`ssl_skip_verify` | `bool` | Skip SSL verification when connecting to Elasticsearch. | | no
+`aliases` | `bool` | Include informational aliases metrics. |  | no
+`data_streams` | `bool` | Export stats for Data Streams. |  | no
+`slm` | `bool` | Export stats for SLM (Snapshot Lifecycle Management). |  | no
+
+
+
+## Exported fields
+The following fields are exported and can be referenced by other components.
+
+Name      | Type                | Description
+--------- | ------------------- | -----------
+`targets` | `list(map(string))` | The targets that can be used to collect `elasticsearch` metrics.
+
+For example, `targets` can either be passed to a `prometheus.relabel`
+component to rewrite the metrics' label set, or to a `prometheus.scrape`
+component that collects the exposed metrics.
+
+The exported targets will use the configured [in-memory traffic][] address
+specified by the [run command][].
+
+[in-memory traffic]: {{< relref "../../concepts/component_controller.md#in-memory-traffic" >}}
+[run command]: {{< relref "../cli/run.md" >}}
+
+## Component health
+
+`prometheus.exporter.elasticsearch` is only reported as unhealthy if given
+an invalid configuration. In those cases, exported fields retain their last
+healthy values.
+
+## Debug information
+
+`prometheus.exporter.elasticsearch` does not expose any component-specific
+debug information.
+
+## Debug metrics
+
+`prometheus.exporter.elasticsearch` does not expose any component-specific
+debug metrics.
+
+## Example
+
+This example uses a [`prometheus.scrape` component][scrape] to collect metrics
+from `prometheus.exporter.elasticsearch`:
+
+```river
+prometheus.exporter.elasticsearch "example" {
+  address = "localhost:9200"
+}
+
+// Configure a prometheus.scrape component to collect Elasticsearch metrics.
+prometheus.scrape "demo" {
+  targets    = prometheus.exporter.elasticsearch.example.targets
+  forward_to = [ /* ... */ ]
+}
```

Comment:
I think this should be as copy-paste-able as possible, so something like:
```suggestion
// Configure a prometheus.scrape component to collect Elasticsearch metrics.
prometheus.scrape "demo" {
  targets    = prometheus.exporter.elasticsearch.example.targets
  forward_to = [ prometheus.remote_write.example.receiver ]
}

prometheus.remote_write "example" {
  // Send metrics to a locally running Mimir.
  endpoint {
    url = "http://mimir:9090/api/v1/write"

    basic_auth {
      username = "sample-username"
      password = "sample-password"
    }
  }
}
```

Reply:
I haven't seen any example like this in the current docs. We could change it, but for now better to be consistent, no?

Reply:
The guide on writing docs hasn't always been followed (and quite a few things predate it), but the guidance is to make it copy-pastable and a full pipeline, when possible: https://github.com/grafana/agent/blob/main/docs/developer/writing-flow-component-documentation.md?plain=1#L494

Reply:
True, I'm not sure there's a stance on exactly _how_ copy-pastable the example needs to be, and it won't be possible to make it work out of the box for every case. But in this case it would be easy for someone to copy-paste the full example into a river config and would just need to replace the url.

It could also be made more generic with something like (which then wouldn't work if basic auth was needed):
```
prometheus.remote_write "example" {
  endpoint {
    url = env("REMOTE_WRITE_URL")
  }
}
```

Reply:
I think your point it's valid, but out of the scope of this PR IMHO. Do you want open issue/PR to address this?

Reply:
Sure, sounds good to me :smile: